security/ssh_certificate.ksy

```diff
@@ -0,0 +1,196 @@
+meta:
+  id: ssh_certificate
+  title: OpenSSH Certificate
+  license: CC0-1.0
+  endian: be
+  imports:
+    - /security/ssh_public_key
+doc: |
+  OpenSSH Certificates are simple certs used by OpenSSH.
+doc-ref: 'https://cvsweb.openbsd.org/cgi-bin/cvsweb/~checkout~/src/usr.bin/ssh/PROTOCOL.certkeys?rev=HEAD&content-type=text/plain'
+seq:
+  - id: cert_type
+    doc: type of ssh certificate
+    type: cstring_utf8
+  - id: body
+    type:
+      switch-on: cert_type.value
+      cases:
+        '"ssh-rsa-cert-v01@openssh.com"': cert_rsa
+        '"ssh-dss-cert-v01@openssh.com"': cert_dss
+        '"ecdsa-sha2-nistp256-cert-v01@openssh.com"': cert_ecdsa
+        '"ecdsa-sha2-nistp384-cert-v01@openssh.com"': cert_ecdsa
+        '"ecdsa-sha2-nistp521-cert-v01@openssh.com"': cert_ecdsa
+        '"ssh-ed25519-cert-v01@openssh.com"': cert_ed25519
+types:
+  cert_rsa:
+    doc-ref: 'https://tools.ietf.org/html/rfc4253#section-6.6'
+    seq:
+      - id: nonce
+        doc: CA-provided random bitstring to prevent hash collisions in the signature
+        type: cstring_bytes
+      - id: rsa
+        type: ssh_public_key::key_rsa
+      - id: footer
+        type: cert_footer
+  cert_dss:
+    seq:
+      - id: nonce
+        doc: CA-provided random bitstring to prevent hash collisions in the signature
+        type: cstring_bytes
+      - id: dsa
+        type: ssh_public_key::key_dsa
+      - id: footer
+        type: cert_footer
+  cert_ecdsa:
+    seq:
+      - id: nonce
+        doc: CA-provided random bitstring to prevent hash collisions in the signature
+        type: cstring_bytes
+      - id: ecdsa
+        type: ssh_public_key::key_ecdsa
+      - id: footer
+        type: cert_footer
+  cert_ed25519:
+    seq:
+      - id: nonce
+        doc: CA-provided random bitstring to prevent hash collisions in the signature
+        type: cstring_bytes
+      - id: ed25519
+        type: ssh_public_key::key_ed25519
+      - id: footer
+        type: cert_footer
+  cert_footer:
+    seq:
+      - id: serial
+        doc: optional serial number; if zero the CA doesn't use serial numbers
+        type: u8
+      - id: type
+        doc: specifies the certificate type (user vs host)
+        type: u4
+        enum: cert_type
+      - id: key_id
+        type: cstring_utf8
+      - id: valid_principals
+        doc: | 
+          As a special case, a zero-length "valid principals"  field means
+          the certificate is valid for any principal of the specified type.
+        type: packed_cstring
+      - id: valid_after
+        doc: Unix timestamp (seconds since 1970-01-01 00:00:00)
+        type: u8
+      - id: valid_before
+        doc: | 
+          Unix timestamp (seconds since 1970-01-01 00:00:00)
+          When "forever" is requested this is set to 0xFFFF_FFFF_FFFF_FFFF
+        type: u8
+      - id: critical_options
+        doc: | 
```

Comment:
```suggestion
        doc: |
          Unix timestamp (seconds since 1970-01-01 00:00:00)
          When "forever" is requested this is set to 0xFFFF_FFFF_FFFF_FFFF
        type: u8
      - id: critical_options
        doc: |
```

There is some trailing whitespace here (and 4 other similar lines). It's recommended to set up your editor to honor the [project `.editorconfig`](https://github.com/kaitai-io/kaitai_struct_formats/blob/1eae5145325bb992e20dae838dd68d04814dda46/.editorconfig#L8) (for example, there is a plugin [_EditorConfig for VS Code_](https://marketplace.visualstudio.com/items?itemName=EditorConfig.EditorConfig)).

Reply:
I'll admit to having done the entire initial go of this in the Kaitai web IDE and the copy/pastaing it to github ;) - I'll go clone and validate that the format and other style guidelines you've called out look to be in order.